docs/api/synthetics/monitors/add-monitor-api.asciidoc

```diff
@@ -0,0 +1,168 @@
+
+:private-locations-url: https://www.elastic.co/guide/en/observability/current/synthetics-private-location.html
+:locations-command: https://www.elastic.co/guide/en/observability/current/synthetics-command-reference.html#elastic-synthetics-locations-command
+
+`locations`:: (https://github.com/elastic/synthetics/blob/main/src/locations/public-locations.ts#L28-L37[`Array<SyntheticsLocationsType>`])
+Where to deploy the monitor. Monitors can be deployed in multiple locations so that you can detect differences in availability and response times across those locations.
++
+To list available locations you can:
++
+  * Run the https://www.elastic.co/guide/en/observability/current/synthetics-command-reference.html#elastic-synthetics-locations-command[`elastic-synthetics locations`] command
+  with the {kib} URL for the deployment from which to fetch available locations.
+  * Go to *Synthetics* -> *Management* and click *Create monitor*.
+    Locations will be listed in _Locations_.
+
+`private_locations` (`Array<string>`)::
+
+The https://www.elastic.co/guide/en/observability/current/synthetics-private-location.html[Private Locations] to which the monitors will be deployed. These Private Locations refer to locations hosted and managed by you, whereas
+`locations` are hosted by Elastic. You can specify a Private Locations using the location's name.
++
+To list available Private Locations you can:
++
+  * Run the https://www.elastic.co/guide/en/observability/current/synthetics-command-reference.html#elastic-synthetics-locations-command[`elastic-synthetics locations`] command
+  with the {kib} URL for the deployment from which to fetch available locations.
+  * Go to *Synthetics* -> *Settings* and click *Private locations*.
+  Private Locations will be listed in the table.
+
+[NOTE]
+====
+You may provide either `locations` or `private_locations`, and you may also provide both if needed. At least one of them is required, but providing both is optional.
+====
+
+
+`enabled`:: (Optional, boolean, default: true): Whether the monitor is enabled.
+
+`tags`:: (Optional, array of strings): An array of tags. 
+
+`alert`:: (Optional, object, default: { status: { enabled: true }, tls: { enabled: true } }): Alert configuration.
+
+`service.name`:: (Optional, string): The APM service name.
+
+`timeout`:: (Optional, string, default: 16): The monitor timeout in seconds.
+
+`namespace`:: (Optional, string, default: "default"): The namespace.
+
+`params`:: (Optional, string): Monitor parameters.
+
+`retryOnFailure`:: (Optional, boolean, default: true): Whether to retry on failure.
+
+*HTTP Monitor Fields*:
+
+`url`:: (Required, string): URL to monitor.
+
+`isTlsEnabled`:: (Optional, boolean, default: false): Whether TLS is enabled.
+
+`max_redirects`:: (Optional, number, default: 0): The maximum number of redirects to follow.
+
+*TCP Monitor Fields*:
+
+`hosts`:: (Required, string): Hosts to monitor.
```

Comment:
While internally it's "hosts" because it matches the heartbeat key, this makes it appear as though we support monitoring multiple hosts via one monitor at this time, which we don't. 

We should use `host` as the public api contract and internally transform it.